### PR TITLE
improve config file documentation

### PR DESCRIPTION
**Pull request by bbernays**

### CODE REVIEW

File: docs/configuration/overview.md [hunk 1]
```diff
@@ -17,22 +20,58 @@ cloudquery:
...
 ```
 
-### Arguments
+### `cloudquery.connection`
 
-* **`connection`** (required) - defines the connections details to your PostgreSQL database.
+A map of values that defines the connections details to your PostgreSQL database.
 
-## Provider Block
+* **`type`** (required) - Type of database that CloudQuery will connect to. Only valid value is `postgres`
```
Comment (disq) [inline docs/configuration/overview.md:29]: `type` is not really required: https://github.com/cloudquery/cloudquery/blob/main/pkg/config/config.go#L116

File: docs/configuration/overview.md [hunk 2]
```diff
@@ -17,22 +20,58 @@ cloudquery:
...
+* **`username`** (required) - Username that CloudQuery will use when interacting with the Postgres database
+* **`password`** (required) - Password for user that CloudQuery will use to authenticate into the Postgres database
+* **`host`** (required) - Hostname or IP address of Postgres database
+* **`port`** (required) - Port of the Postgres database that CloudQuery will connect to
+* **`database`** (required) - Name of the Postgres database that CloudQuery will connect to
+* **`sslmode`** (required) - Postgres setting for specifying the level of security you want to enforce in the connection between CloudQuery and your database. If you are running CloudQuery locally in a docker container the typical value is `disable`. Other valid options include: `allow`, `prefer`, `require`, `verify-ca`, `verify-full`
```
Comment (disq) [inline docs/configuration/overview.md:35]: Not required.

**Review by disq — APPROVED** [review]
LGTM with nits

File: docs/configuration/overview.md (hunk 2 above)
Comment (bbernays) [reply docs/configuration/overview.md:35]: How can all 4 of those not be required?

Comment (bbernays) [reply docs/configuration/overview.md]: How do we set the defaults?

File: docs/configuration/overview.md (hunk 1 above)
Comment (bbernays) [reply docs/configuration/overview.md]: Changed to `optional`

File: docs/configuration/overview.md [hunk 3]
```diff
@@ -17,22 +20,58 @@ cloudquery:
...
-* **`connection`** (required) - defines the connections details to your PostgreSQL database.
+A map of values that defines the connections details to your PostgreSQL database.
 
-## Provider Block
+* **`type`** **(required)** - Type of database that CloudQuery will connect to. Only valid value is `postgres`
+* **`username`** **(required)** - Username that CloudQuery will use when interacting with the Postgres database
+* **`password`** **(required)** - Password for user that CloudQuery will use to authenticate into the Postgres database
+* **`host`** **(required)** - Hostname or IP address of Postgres database
+* **`port`** **(optional)** - Port of the Postgres database that CloudQuery will connect to. Default value is `5432`
+* **`database`** (required)** - Name of the Postgres database that CloudQuery will connect to
```
Comment (bbernays) [inline docs/configuration/overview.md:34]:
```suggestion
* **`database`** **(required)** - Name of the Postgres database that CloudQuery will connect to
```

**Pull request merged**